packages/docs/getting-started/running-a-validator.md

```diff
@@ -46,33 +66,227 @@ The code you'll see on this page is bash commands and their output.
+```
+
+Now we can start up the validator node:
+
+```bash
+docker run --name celo-validator -p 127.0.0.1:8547:8545 -p 127.0.0.1:8548:8546 -p 30304:30303 -p 30304:30303/udp -v $PWD/validator:/root/.celo $CELO_IMAGE:$CELO_NETWORK --verbosity 3 --networkid $NETWORK_ID --syncmode full --rpc --rpcaddr 0.0.0.0 --rpcapi eth,net,web3,debug --maxpeers 125 --mine --istanbul.blockperiod=1 --istanbul.requesttimeout=3000 --etherbase $CELO_VALIDATOR_ADDRESS --nodiscover --proxy.proxied --proxy.proxyenodeurlpair=enode://$PROXY_ENODE@$PROXY_IP:30503\;enode://$PROXY_ENODE@$PROXY_IP:30503
+```
+
+{% hint style="danger" %}
+**Security**: The command line above includes the parameter `--rpcaddr 0.0.0.0` which makes the Celo Blockchain software listen for incoming RPC requests on all network adaptors. Exercise extreme caution in doing this when running outside Docker, as it means that any unlocked accounts and their funds may be accessed from other machines on the Internet. In the context of running a Docker container on your local machine, this together with the `docker -p` flags allows you to make RPC calls from outside the container, i.e from your local host, but not from outside your machine. Read more about [Docker Networking](https://docs.docker.com/network/network-tutorial-standalone/#use-user-defined-bridge-networks) here.
+{% endhint %}
+
+The `mine` flag does not mean the node starts mining blocks, but rather starts trying to participate in the BFT consensus protocol. It cannot do this until it gets elected -- so next we need to stand for election.
+
+The `networkid` parameter value of `44785` indicates we are connecting the Baklava Beta network.
+
+#### Deploy the attestation service
+
+As part of the [lightweight identity protocol](/celo-codebase/protocol/identity), validators are expected to run an attestation service to provide attestations that allow users to map their phone number to an account on Celo. The attestation service is a simple Node.js application that can be run with a docker image:
+
+First we need to create an account. Run:
+
+```bash
+celocli account:new
+```
+
+We copy the account details and assign the Private Key to the `ATTESTATION_SERVICE` environment variable:
+
+```bash
+export ATTESTATION_KEY=<Private Key>
+```
+
+```bash
+docker run -e ATTESTATION_KEY=$ATTESTATION_KEY -p 3000:80 us.gcr.io/celo-testnet/attestation-service:$CELO_NETWORK
+```
+
+In order for users to request attestations from your service, you need to register the endpoint under which your service is reachable in your [metadata](/celo-codebase/protocol/identity/metadata).
+
+```bash
+celocli identity:create-metadata ./metadata.json
+```
+
+Add your URL:
+
+```bash
+celocli identity:change-attestation-service-url ./metadata.json --url ATTESTATION_SERVICE_URL
+```
+
+And then host your metadata somewhere reachable via HTTP. You can register your metadata URL with:
```

Comment:
might be a good idea to suggest a gist and to use the url to the raw text

Reply:
:+1: